velox/functions/sparksql/tests/ArrayMinTest.cpp

```diff
@@ -0,0 +1,223 @@
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class ArrayMinTest : public SparkFunctionBaseTest {
+ protected:
+  template <typename T>
+  void testExpr(
+      const VectorPtr& expected,
+      const std::string& expression,
+      const std::vector<VectorPtr>& input) {
+    auto result = evaluate<SimpleVector<T>>(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+  }
+
+  template <typename T>
+  void testIntNullable() {
+    auto arrayVector = makeNullableArrayVector<T>(
+        {{-1, 0, 1, 2, 3, 4},
+         {4, 3, 2, 1, std::nullopt, 0, -1, -2},
+         {-5, -4, -3, -2, -1},
+         {101, 102, 103, 104, std::nullopt},
+         {std::nullopt, -1, -2, std::nullopt, -3, -4},
+         {},
+         {std::nullopt, std::nullopt}});
+    auto expected = makeNullableFlatVector<T>(
+        {-1, -2, -5, 101, -4, std::nullopt, std::nullopt});
+    testExpr<T>(expected, "array_min(C0)", {arrayVector});
+  }
+
+  template <typename T>
+  void testInt() {
+    auto arrayVector = makeArrayVector<T>(
+        {{-1, 0, 1, 2, 3, 4},
+         {4, 3, 2, 1, 0, -1, -2},
+         {-5, -4, -3, -2, -1},
+         {101, 102, 103, 104, 105},
+         {}});
+    auto expected = makeNullableFlatVector<T>({-1, -2, -5, 101, std::nullopt});
+    testExpr<T>(expected, "array_min(C0)", {arrayVector});
+  }
+
+  void testDateNullable() {
+    auto D = [](const std::string& dateStr) { return DATE()->toDays(dateStr); };
+    auto arrayVector = makeNullableArrayVector<int32_t>(
+        {{D("1970-01-01"), D("2023-08-23")},
+         {},
+         {D("1970-01-01"), std::nullopt}},
+        ARRAY(DATE()));
+    auto expected = makeNullableFlatVector<int32_t>(
+        {D("1970-01-01"), std::nullopt, D("1970-01-01")}, DATE());
+    testExpr<int32_t>(expected, "array_min(C0)", {arrayVector});
+  }
+
+  void testDate() {
+    auto D = [](const std::string& dateStr) { return DATE()->toDays(dateStr); };
+    auto arrayVector = makeArrayVector<int32_t>(
+        {{D("1970-01-01"), D("2023-08-23")}, {}}, DATE());
+    auto expected = makeNullableFlatVector<int32_t>(
+        {D("1970-01-01"), std::nullopt}, DATE());
+    testExpr<int32_t>(expected, "array_min(C0)", {arrayVector});
+  }
+
+  void testInLineVarcharNullable() {
+    using S = StringView;
+
+    auto arrayVector = makeNullableArrayVector<S>({
+        {S("red"), S("blue")},
+        {std::nullopt, S("blue"), S("yellow"), S("orange")},
+        {},
+        {std::nullopt, std::nullopt},
+        {S("red"), S("purple"), S("green")},
+    });
+    auto expected = makeNullableFlatVector<S>(
+        {S("blue"), S("blue"), std::nullopt, std::nullopt, S("green")});
+    testExpr<S>(expected, "array_min(C0)", {arrayVector});
+  }
+
+  void testVarcharNullable() {
+    using S = StringView;
```

Comment:
This is not needed. Use "red shiny car ahead"_s instead.

Reply:
Also, I think you can use makeNullableArrayVector<std::string>(..."red shiny car ahead"...)

Reply:
improved with the second method.